test/graphql/schemas/args/argGenerator.test.ts

```diff
@@ -0,0 +1,186 @@
+import "reflect-metadata";
+import { beforeEach, describe, expect, it } from "vitest";
+
+import { Field, ObjectType } from "type-graphql";
+import {
+  createEntityArgs,
+  typeCache,
+} from "../../../../src/graphql/schemas/args/argGenerator.js";
+import { SortOrder } from "../../../../src/graphql/schemas/enums/sortEnums.js";
+
+@ObjectType()
+class ReferencedEntity {
+  @Field(() => String)
+  id!: string;
+
+  @Field(() => Number)
+  count!: number;
+}
+// Test entities
+@ObjectType()
+class TestEntity {
+  @Field(() => String)
+  id!: string;
+
+  @Field(() => String)
+  name!: string;
+
+  @Field(() => TestEntity, { nullable: true })
+  nested?: TestEntity;
+
+  @Field(() => ReferencedEntity, { nullable: true })
+  reference?: ReferencedEntity;
+}
+
+describe("argGenerator", () => {
+  describe("WhereArgs", () => {
+    beforeEach(() => {
+      Object.keys(typeCache).forEach((key) => {
+        // eslint-disable-next-line @typescript-eslint/no-dynamic-delete, @typescript-eslint/no-explicit-any
+        delete (typeCache as any)[key];
+      });
+    });
+
+    it("should create basic where args with all fields", () => {
+      const { WhereArgs } = createEntityArgs<TestEntity>("Test", {
+        id: "id",
+        name: "string",
+        reference: {
+          type: "id",
+          references: {
+            entity: ReferencedEntity,
+            fields: {
+              count: "number",
+            },
+          },
+        },
+      });
+
+      const instance = new WhereArgs();
+      expect(Object.keys(instance)).toContain("id");
+      expect(Object.keys(instance)).toContain("name");
+      expect(Object.keys(instance)).toContain("reference");
+
+      // Test field assignments
+      instance.id = { eq: "123" };
```

Comment:
Can you explain this part of the test case? Why do you need to test if assignment works?

Reply:
It's no longer present, but I think this was an artifact from us not actually doing any type checking and me using this as smoke tests. By now, we use `tsc` in the build step.